rust/src/storage/s3/dynamodb_lock.rs

```diff
@@ -439,8 +462,14 @@ struct AcquireLockState<'a> {
 }
 
 impl<'a> AcquireLockState<'a> {
+    /// If lock is expirable (lease_duration is set) and is do_not_timeout_on_non_expirable_locks
+    /// then this function returns `true` if the elapsed time sine `started` is reached `timeout_in`.
     fn has_timed_out(&self) -> bool {
-        self.started.elapsed() > self.timeout_in
+        self.started.elapsed() > self.timeout_in && {
+            let is_non_expirable = self.cached_lock.is_some()
+                && self.cached_lock.as_ref().unwrap().lease_duration.is_none();
```

Comment:
have you tried this?

```rust
let non_expirable = if let Some(cached_lock) = self.cached_lock {
    cached_lock.lease_duration.is_none()
} else {
    false
};
```

Reply:
Yes, I figured that this one results on more code, however if that's preferable convention I'll stick with that

Reply:
yeah, i think this is slightly better because it avoids a redundant enum match.

Reply:
Oh I see, ok